.github/workflows/rack-box-continuous.yml

```diff
@@ -1,11 +1,16 @@
-# Builds a Docker image upon each push to master:
+# Builds a Docker image upon every push (while ignoring tag-only pushes):
 #  - Caches files needed to build rack-box images
 #  - Builds rack-box:dev Docker image
 #  - Pushes rack-box:dev image to Docker Hub
 
 name: RACK-in-a-Box continous build
 
-on: [push]
+on:
+  pull_request:
+    branches: [ '*' ]
+  push:
+    branches: [ '*' ]
+    tags-ignore: [ '*' ]
```

Comment:
Now we will run our rack-box checks on both pull requests and pushes while ignoring tag-only pushes (hence avoiding starting redundant workflows when publishing a release).

Reply:
Nice, avoiding redundant workflows sounds great. Do you know what the difference would be between the following two configurations?
```yaml
on:
  push:
    branches: [ '*' ]
```
vs
```yaml
on:
  pull_request:
    branches: [ '*' ]
  push:
    branches: [ '*' ]
```
Specifically, I'm curious if the second causes two builds for every push to a branch that also has an open PR.

Reply:
Following up on this: It looks like the semantics of this are that we only build on _pushes to branches that have an associated pull request_ which is a bit more restrictive than we had previously. In other words, the conditions in the two stanzas are ANDed together. 

@tuxji What are your thoughts on this configuration? I personally prefer running builds on pushes to _any_ branch, so that I can be sure my changes are passing CI before opening a pull request and notifying everyone who is watching the repo.

Reply:
I didn't know that the conditions in the two stanzas are AND'ed together.  I've seen other workflows that say:

`on: [pull_request, push]`

Can you find some GitHub documentation which states that `on: [pull_request, push]` is more restrictive than `on: pull_request` and `on: push` evaluated separately?  For that matter, I've also seen workflows which add `workflow_dispatch:` to these conditions and these workflows can be started from the GitHub GUI without making a git push or pull request as far as I understand.

Reply:
Huh, the documentation says

> ```yaml
> # Triggers the workflow on push or pull request events
> on: [push, pull_request]
> ```

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-a-list-of-events

I thought that I saw the AND behavior, but perhaps I was mistaken. Let's leave this for now and I'll gather more examples next time I see something that doesn't match my expectations.